### PR TITLE
add dwm

### PR DESCRIPTION
**Pull request by asdfugil**
well, that took some time to figure out
remmina_iPad_192 168 0 102:5901_202157-34431 014265

### CODE REVIEW

File: dwm.mk [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	$(SED) -i '3 a #include <TargetConditionals.h>' $(BUILD_WORK)/dwm/drw.c $(BUILD_WORK)/dwm/dwm.c
+	$(SED) -i 's/{ "st", NULL };/{ "xterm", "-bg", "black", "-fg", "lightgray", NULL };/g' $(BUILD_WORK)/dwm/config.def.h
+	$(SED) -i 's/#005577/#005377/g' $(BUILD_WORK)/dwm/config.def.h
+
+ifneq ($(wildcard $(BUILD_WORK)/dwm/.build_complete),)
+dwm:
+	@echo "Using previously built dwm."
+else
+dwm: libx11 libxft fontconfig freetype dwm-setup
+	cd $(BUILD_WORK)/dwm; \
```
Comment (Gymxo) [inline dwm.mk:29]: What's the point of this cd here?

**Review by Gymxo — CHANGES_REQUESTED** [review]

File: dwm.mk (hunk 1 above)
Comment (asdfugil) [reply dwm.mk:29]: no point

**Comment by asdfugil** [comment]
Don't merge yet because I want to enable libxinerama

**Comment by CRKatri** [comment]
How do you start this? Do we have xinit?

**Comment by Gymxo** [comment]
In this case it’s probably dwm hence the name

**Comment by CRKatri** [comment]
I know that but xinit is a cleaner way, I just didn’t know if pro had it

**Comment by Gymxo** [comment]
I can send a pr if wanted since it’s basically almost done on my end

File: dwm.mk [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	wget -q -nc -P$(BUILD_SOURCE) https://dl.suckless.org/dwm/dwm-6.2.tar.gz
+	$(call EXTRACT_TAR,dwm-$(DWM_VERSION).tar.gz,dwm-$(DWM_VERSION),dwm)
+	$(SED) -i 's@/usr/local@$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)@g' $(BUILD_WORK)/dwm/config.mk
+	$(SED) -i 's@cc@$(CC)@g' $(BUILD_WORK)/dwm/config.mk
+	$(SED) -i 's@/usr/include@$(BUILD_BASE)$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/include@g' $(BUILD_WORK)/dwm/config.mk
+	$(SED) -i 's@/usr/X11R6/include@$(BUILD_BASE)$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/include@g' $(BUILD_WORK)/dwm/config.mk
+	$(SED) -i 's@/usr/X11R6/lib@$(BUILD_BASE)$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib@g' $(BUILD_WORK)/dwm/config.mk
+	$(SED) -i 's@snprintf@// snprintf@g' $(BUILD_WORK)/dwm/dwm.c
+	$(SED) -i '3 a #include <TargetConditionals.h>' $(BUILD_WORK)/dwm/drw.c $(BUILD_WORK)/dwm/dwm.c
+	$(SED) -i 's/{ "st", NULL };/{ "xterm", "-bg", "black", "-fg", "lightgray", NULL };/g' $(BUILD_WORK)/dwm/config.def.h
```
Comment (m1stadev) [inline dwm.mk:19]: Move these sed calls to a patch.

**Review by m1stadev — CHANGES_REQUESTED** [review]

**Comment by asdfugil** [comment]
@m1stadev looks good to me

**Review by Gymxo — APPROVED** [review]

File: makefiles/dwm.mk [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	$(MAKE) -C $(BUILD_WORK)/dwm
+	+$(MAKE) -C $(BUILD_WORK)/dwm install \
+		DESTDIR=$(BUILD_STAGE)/dwm
+	$(call AFTER_BUILD)
+endif
+
+dwm-package: dwm-stage
+	# dwm.mk Package Structure
+	rm -rf $(BUILD_DIST)/dwm
+	mkdir -p $(BUILD_DIST)/dwm
```
Comment (Gymxo) [inline makefiles/dwm.mk:28]: Remove line 28 of the dwm.mk

Comment (asdfugil) [reply makefiles/dwm.mk:28]:
```suggestion
```

Comment (Gymxo) [reply makefiles/dwm.mk]: Yeah I hate the ui on mobile safari

**Review by Gymxo — APPROVED** [review]

**Comment by asdfugil** [comment]
By the way, I did not have a breakdown - I just can't allow myself to be distracted while I do something much more important and takes time.

**Comment by asdfugil** [comment]
The current ETA is afternoon of 6 May 2022 (UTC+8), but is prone to delays.

**Comment by asdfugil** [comment]
`build_tools/compile.sh` works fine. Also CFLAGS is now properly set as opposed to using -Wno-error-implicit-function-declaration

**Comment by asdfugil** [comment]
bingo

**Pull request closed without merging**